airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresType.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.postgres;
+
+import java.sql.SQLType;
+import java.sql.Types;
+
+public enum PostgresType implements SQLType {
```

Comment:
Am just concerned that if we ever upgrade the JDBC driver, these types might change and now we will have to maintain our own version of all the types and we might miss out on the changes

Reply:
> Am just concerned that if we ever upgrade the JDBC driver, these types might change and now we will have to maintain our own version of all the types and we might miss out on the changes

@subodh1810 very good point. At the moment, the postgres source in master branch uses JDBCType, which is included in the java.sql package, so even now we cannot handle all the data types that postgres has. When updating the jdbc driver, the postgres data types may change and we still won’t be able to map them to JDBCType correctly
PostgresType is just copy-paste JDBCType + postgres arrays

I didn't find any other way to add new data types other than adding PostgresType as JDBCType is generic for most databases and doesn't support any custom types
Do you have any advice on how to map custom postgres data types?